app/javascript/src/app/controllers/checkbox_tree_branch_controller.js

```diff
@@ -0,0 +1,41 @@
+import { Controller } from 'stimulus';
+
+export default class extends Controller {
+  static targets = ['toggle', 'selectAll', 'leaves', 'leaf'];
+
+  toggleBranch(event) {
+    this.toggleTarget.classList.toggle('open');
+    this.leavesTarget.classList.toggle('open');
+  }
+
+  toggleSelectAll() {
+    this.leafTargets.forEach(leaf => {
+      leaf.checked = this.selectAllTarget.checked;
+    });
+  }
+
+  handleLeafChange() {
+    const allChecked = () => {
+      return this.leafTargets.every((leaf) => {
+        return leaf.checked;
+      });
+    };
+
+    const anyChecked = () => {
+      return this.leafTargets.some((leaf) => {
+        return leaf.checked;
+      });
+    }
```

Comment:
I might consider moving these helper methods out as first class _private_ methods of the controller.